system_setup/common/wsl_utils.py

```diff
@@ -92,3 +94,74 @@ def get_windows_locale():
     except OSError as e:
         log.info(windows_locale_failed_msg + e.strerror)
         return None
+
+
+def get_exec_from_distro_name(distro_name):
+    """
+    Return the executable name for the given distro name.
+
+    :param distro_name: The distro name.
+    """
+    proc1 = \
+        run_command(["reg.exe", "query",
+                     "HKCU\\SOFTWARE\\Microsoft\\"
+                     "Windows\\CurrentVersion\\Lxss",
+                     "/s", "/f", "DistributionName"])
+    if proc1.returncode:
+        log.info("failed to get distro name: "
+                 + proc1.stderr)
+        return None
+    raw_data = proc1.stdout[2:]
+    raw_list = raw_data.split("\r\n\r\n")[:-1]
+    for raw_item in raw_list:
+        if raw_item.endswith(distro_name):
+            exec_loc = raw_item.split("\r\n")[0]
+            proc2 = run_command(["reg.exe", "query",
+                                 exec_loc, "/v", "PackageFamilyName"])
+            if proc2.returncode:
+                log.info("failed to get package name: "
+                         + proc2.stderr)
+                return None
+            pkg_name = proc2.stdout.replace("\r\n", "").split()[-1]
+            proc3 = \
+                run_command(["powershell.exe", "-NoProfile",
+                             "-NonInteractive", "-Command",
+                             "[Environment]::GetFolderPath"
+                             "('LocalApplicationData')"])
+            if proc3.returncode:
+                log.info("failed to get LocalApplicationData: "
+                         + proc3.stderr)
+                return None
+            raw_win_path = proc3.stdout.replace("\r\n", "")
+            win_path = raw_win_path + "\\Microsoft\\WindowsApps\\" + pkg_name
+            proc4 = run_command(["wslpath", win_path])
+            if proc4.returncode:
+                log.info("failed to get convert path: "
+                         + proc4.stderr)
+                return None
+            exe_real_loc = proc4.stdout.replace("\n", "")
+            return os.listdir(exe_real_loc)[0]
+    return None
+
+
+def perform_wsl_shutdown(is_restart=False):
+    shutdown_cmd = ("Start-Process -FilePath 'powershell'"
+                    " -ArgumentList '-NoProfile','-NonInteractive',"
+                    "'-Command','wsl.exe','--shutdown',';'")
+    if is_restart:
+        if os.environ.get('WT_PROFILE_ID') is not None:
+            shutdown_cmd = shutdown_cmd + ",'wt.exe'"
+
+        exec_name = ",'wsl.exe'"
+        if os.environ.get('WSL_DISTRO_NAME') is not None:
+            wsl_distro_name = os.environ.get('WSL_DISTRO_NAME')
+            exec_name = exec_name + ",'-d','" + wsl_distro_name + "'"
```

Comment:
Can you format a string with .format()? That would make things easier to read (same for other string management below)